sql/expression/plus.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dolthub/vitess/go/vt/sqlparser"
+	"github.com/shopspring/decimal"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+var _ ArithmeticOp = (*Plus)(nil)
+var _ sql.CollationCoercible = (*Plus)(nil)
+
+// Plus expressions include plus, minus and multiplication (+, -, *) operations.
+type Plus struct {
+	BinaryExpression
+	ops int32
+}
+
+// NewPlus creates a new Plus sql.Expression.
+func NewPlus(left, right sql.Expression) *Plus {
+	a := &Plus{BinaryExpression{Left: left, Right: right}, 0}
+	ops := countArithmeticOps(a)
+	setArithmeticOps(a, ops)
+	return a
+}
+
+func (p *Plus) LeftChild() sql.Expression {
+	return p.Left
+}
+
+func (p *Plus) RightChild() sql.Expression {
+	return p.Right
+}
+
+func (p *Plus) Operator() string {
+	return sqlparser.PlusStr
+}
+
+func (p *Plus) SetOpCount(i int32) {
+	p.ops = i
+}
+
+func (p *Plus) String() string {
+	return fmt.Sprintf("(%s + %s)", p.Left, p.Right)
+}
+
+func (p *Plus) DebugString() string {
+	return fmt.Sprintf("(%s + %s)", sql.DebugString(p.Left), sql.DebugString(p.Right))
+}
+
+// IsNullable implements the sql.Expression interface.
+func (p *Plus) IsNullable() bool {
+	if types.IsDatetimeType(p.Type()) || types.IsTimestampType(p.Type()) {
+		return true
+	}
+
+	return p.BinaryExpression.IsNullable()
+}
+
+// Type returns the greatest type for given operation.
+func (p *Plus) Type() sql.Type {
```

Comment:
This function looks like it's the same as the `Type()` function for the new `Minus` type, too. It would be helpful to have them reuse the same helper function, instead of having this code copied, so that when we fix any issues in one implementation, we don't have to remember to fix it in a second spot.